addons/viewport/src/Tool.js

```diff
@@ -1,121 +1,120 @@
-import { document } from 'global';
 import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
 import memoize from 'memoizerific';
 
-import { Global, css } from '@storybook/theming';
+import { Global } from '@storybook/theming';
 
 import { Popout, Item, Icons, Icon, IconButton, Title, List } from '@storybook/components';
-import { STORY_CHANGED } from '@storybook/core-events';
-import { logger } from '@storybook/client-logger';
+import { SET_STORIES } from '@storybook/core-events';
 
 import { PARAM_KEY } from './constants';
 
-const toList = memoize(50)(viewports => Object.entries(viewports));
-const getIframe = memoize(1)(() => document.getElementById('storybook-preview-iframe'));
-const iframeClass = 'storybook-preview-iframe-viewport';
+const toList = memoize(50)(items =>
+  items ? Object.entries(items).map(([id, value]) => ({ ...value, id })) : []
+);
+const iframeId = 'storybook-preview-background';
+
+const getState = memoize(10)((props, state) => {
+  const data = props.api.getCurrentStoryData();
+  const list = toList(data && data.parameters && data.parameters[PARAM_KEY]);
+
+  return list && list.length
+    ? list.reduce(
+        (acc, { name, styles: value, id }) => {
+          acc.items.push({ name, value, id });
+
+          if (state.selected !== 'responsive') {
+            if (!list.find(i => i.id === state.selected)) {
+              acc.selected = id;
```

Comment:
I don't understand this code. Does it keep setting `acc.selected` if `state.selected` isn't in the list? It seems like it'll do this for every item in the list.

I think this code is overly complex for what it is trying to do -- or maybe I don't understand it? In either case I think it needs to be clearer. I *think* this is what it shoudl probably look like:

```js
const items = list && list.length && list.map(({ name, styles: value, id }) => ({name, value, id});

// It actually will be the *last* element in the list in the code above I think, but I doubt this is what you meant?
const selected = (state.selected === 'responsive' || list.find(i => i.id === state.selected))  ? state.selected : list[0].id;

return {
  list,
  selected,
  isRotated: state.isRotated,
};
```